lib/Screens/profile_screen.dart

```diff
@@ -0,0 +1,113 @@
+import 'dart:convert';
+import 'package:basic_authentication_flutter_challenge/Screens/login_page.dart';
+import 'package:basic_authentication_flutter_challenge/Screens/all_users.dart';
+import 'package:flutter/material.dart';
+
+
+
+
+class ProfilePage extends StatefulWidget {
+ 
+  const ProfilePage({ Key? key}) : super(key: key);
+
+  @override
+  State<ProfilePage> createState() => _ProfilePageState();
+}
+
+class _ProfilePageState extends State<ProfilePage> {
+
+LoginPage user= LoginPage();
+  String fname = '';
+  String lname = '';
+  String email = '';
+  String username = '';
+  String company = '';
+  String gender = '';
+
+
+
+    void getInfo() async {
+      final jsonInfo=jsonDecode(await user.api.userAuth());
```

Comment:
You extracted data of logged in user very well. Try to look for json serialization to learn how to convert the received json response to an object that you can deal with it easily.